app/routes/thematic/geoencoder.py

```diff
@@ -0,0 +1,216 @@
+import re
+from typing import Any, Dict, List, Optional
+
+from fastapi import APIRouter, HTTPException, Query
+from unidecode import unidecode
+
+from app.crud.versions import get_version, get_version_names
+from app.errors import RecordNotFoundError
+from app.models.pydantic.responses import Response
+from app.routes import VERSION_REGEX
+from app.routes.datasets.queries import _query_dataset_json
+
+router = APIRouter()
+
+
+@router.get(
+    "/geoencode",
+    tags=["Geoencoder"],
+    status_code=200,
+)
+async def geoencode(
+    *,
+    admin_source: str = Query(
+        "GADM", description="The source of administrative boundaries to use."
+    ),
+    admin_version: str = Query(
+        ...,
+        description="Version of the administrative boundaries dataset to use.",
```

Comment:
Similar, is there a way to document the choices available? I guess this may get more confusing if ever have multiple providers

Reply:
Wild idea: should we consolidate admin version and admin dataset to one field, and have the options be like: "GADM 3.6", "GADM 4.1", "geoBoundaries 1.0", "middleEarth 3.2". Then it'll be clear what you're getting from a set of options.

Reply:
Combining the provider and version has merit (`middleEarth 3.2`). It has excellent cohesion.
This is such a simple scenario that it's hard to justify breaking it up.

However, we'd be encoding a string with special information (i.e., _provider_`space`_version_).
It's generally a good practice to split information up even though it makes for a bit more complexity.
It's more flexible in the long run (although I'm starting to violate YAGNI with this statement).
Finally, we are splitting this information up in the area/geostore microservice interface for all of the reasons mentioned above. 

That's my two cents.